bounty_hunter/src/models.rs

```diff
@@ -1,9 +1,277 @@
-use super::schema::status;
-
-#[derive(Queryable, Serialize, Deserialize, Debug, Insertable)]
-#[table_name = "status"]
-pub struct Status {
-    pub ip: String,
-    pub mac: String,
-    pub balance: String,
+use clarity::Signature;
+use failure::Error;
+use num256::Uint256;
+use num_traits::Zero;
+
+use std::convert::From;
+
+use schema::states;
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+pub struct ChannelState {
+    pub channel_id: Uint256,
+    pub nonce: Uint256,
+
+    pub balance_a: Uint256,
+    pub balance_b: Uint256,
+
+    pub signature_a: Option<Signature>,
+    pub signature_b: Option<Signature>,
+}
+
+/// A helper type that prepares a [ChannelState](ChannelState) for storage by turning the contents into types
+/// easy to grasp for `diesel`.
+#[derive(Queryable, Debug, Clone, PartialEq, Eq, Default)]
+pub struct ChannelStateRecord {
+    pub id: i32,
+    pub channel_id: String,
+    pub nonce: String,
+
+    pub balance_a: String,
+    pub balance_b: String,
+
+    pub sig_a_v: Option<String>,
+    pub sig_a_r: Option<String>,
+    pub sig_a_s: Option<String>,
+
+    pub sig_b_v: Option<String>,
+    pub sig_b_r: Option<String>,
+    pub sig_b_s: Option<String>,
+}
+
+#[derive(Insertable, Clone, Debug, Eq, PartialEq)]
+#[table_name = "states"]
+pub struct NewChannelStateRecord {
+    pub channel_id: String,
+    pub nonce: String,
+
+    pub balance_a: String,
+    pub balance_b: String,
+
+    pub sig_a_v: Option<String>,
+    pub sig_a_r: Option<String>,
+    pub sig_a_s: Option<String>,
+
+    pub sig_b_v: Option<String>,
+    pub sig_b_r: Option<String>,
+    pub sig_b_s: Option<String>,
+}
+
+impl Default for ChannelState {
+    fn default() -> Self {
+        Self {
+            channel_id: Uint256::zero(),
+            nonce: Uint256::zero(),
+            balance_a: Uint256::zero(),
+            balance_b: Uint256::zero(),
+            signature_a: None,
+            signature_b: None,
+        }
+    }
+}
+
+impl From<ChannelStateRecord> for NewChannelStateRecord {
+    fn from(record: ChannelStateRecord) -> Self {
+        NewChannelStateRecord {
+            channel_id: record.channel_id,
+            nonce: record.nonce,
+
+            balance_a: record.balance_a,
+            balance_b: record.balance_b,
+
+            sig_a_v: record.sig_a_v,
```

Comment:
I'm almost sure we can store sigs as their hexadecimal representations (or even better - a binary form or even `Signature` types if they could be converted to-from their database forms on the fly) That might make serialization/deserialization easier - so instead of 6 nullable fields you can have just two nullable fields (sig_a, sig_b). We can add `FromStr` in Signature that would parse it, and `to_string` already exists. Thoughts?

Reply:
Converted everything to `Vec<u8>`s via `to_bytes_be` and `from_bytes_be`.